src/main/java/com/amihaiemil/camel/NoCommentsYamlLine.java

```diff
@@ -56,22 +56,21 @@ public int compareTo(final YamlLine other) {
 
     @Override
     public String trimmed() {
-        String string = this.line.trimmed();
+        String trimmed = this.line.trimmed();
         int i = 0;
-        while(i < string.length()) {
-            if(string.charAt(i) == '#') {
-                string = string.substring(0, i);
+        while(i < trimmed.length()) {
+            if(trimmed.charAt(i) == '#') {
+                trimmed = trimmed.substring(0, i);
                 break;
-            } else if(string.charAt(i) == '"') {
+            } else if(trimmed.charAt(i) == '"') {
                 i++;
-                while(i < string.length() && string.charAt(i) != '"') {
+                while(i < trimmed.length() && trimmed.charAt(i) != '"') {
                     i++;
                 }
-            } else {
-                i++;
             }
+            i++;
```

Comment:
@SherifWaly Well, I tried now the code, and the test without the ``else`` and without the modification passes fine. How did it fail for you? :D (you said you modified it because it failed once you removed the ``else``).

After running the code, my opinion is that it's correct without the ``else`` and also the original test passes. 
It's correct without the else because the ``while`` in ``else if`` stops at a ``"`` - then you need to make ``i++`` to get to the next char :D